### PR TITLE
bpo-38493: Add os.CLD_KILLED and  CLD_STOPPED  for os

### PR DESCRIPTION
**Pull request by corona10**
https://bugs.python.org/issue38493

### CODE REVIEW

File: Doc/library/os.rst [hunk 1]
```diff
@@ -3946,6 +3946,7 @@ written in Python, such as a mail server's external command delivery program.
 
 
 .. data:: CLD_EXITED
+          CLD_KILLED
```
Comment (serhiy-storchaka) [inline Doc/library/os.rst:3949]: What about CLD_STOPPED?

Please add also the `versionchanged` directive.

File: Misc/NEWS.d/next/Library/2019-10-16-19-56-51.bpo-38373.86ExWB.rst [hunk 2]
```diff
@@ -0,0 +1,2 @@
+:data:`~os.CLD_KILLED` is added for posixmodule.
```
Comment (serhiy-storchaka) [inline Misc/NEWS.d/next/Library/2019-10-16-19-56-51.bpo-38373.86ExWB.rst]: ``` :mod:`os` ```

File: Misc/NEWS.d/next/Library/2019-10-16-19-56-51.bpo-38373.86ExWB.rst [hunk 3]
```diff
@@ -0,0 +1,2 @@
...
+(Contributed by Dong-hee Na in :issue:`38493`.)
```
Comment (serhiy-storchaka) [inline Misc/NEWS.d/next/Library/2019-10-16-19-56-51.bpo-38373.86ExWB.rst]: `Patch by Dong-hee Na.`

The issue number is already specified in the file name.

Please add a What's New entry for this change (and there the `Contributed by` will be appropriate).

File: Doc/library/os.rst (hunk 1 above)
Comment (corona10) [reply Doc/library/os.rst:3949]: @serhiy-storchaka Oh thanks for letting me know.

File: Lib/test/test_posix.py [hunk 4]
```diff
@@ -1226,6 +1226,18 @@ def test_rename_dir_fd(self):
         finally:
             posix.close(f)
 
+    @unittest.skipUnless(hasattr(signal, 'SIGCHLD'), 'CLD_XXXX be placed in si_code for a SIGCHLD signal')
+    @unittest.skipUnless(hasattr(os, 'waitid_result'), "test needs os.waitid_result")
+    def test_cld_xxxx_constants(self):
+        # reference: http://man7.org/linux/man-pages/man2/sigaction.2.html
```
Comment (serhiy-storchaka) [inline Lib/test/test_posix.py]: These names are not Linux specific, and on other OSes they can have other values.

File: Lib/test/test_posix.py [hunk 5]
```diff
@@ -1226,6 +1226,18 @@ def test_rename_dir_fd(self):
...
+        self.assertEqual(os.CLD_EXITED, 1)
+        self.assertEqual(os.CLD_KILLED, 2)
+        self.assertEqual(os.CLD_DUMPED, 3)
+        self.assertEqual(os.CLD_TRAPPED, 4)
+        self.assertEqual(os.CLD_STOPPED, 5)
+        # FIXME: CLD_CONTINUED was added since Linux 2.6.9
```
Comment (serhiy-storchaka) [inline Lib/test/test_posix.py]: I do not think Python 3.9 supports Linux older than 2.6.9.

Comment (corona10) [reply Lib/test/test_posix.py]: Got it!

File: Lib/test/test_posix.py (hunk 4 above)
Comment (corona10) [reply Lib/test/test_posix.py]: > These names are not Linux specific, and on other OSes they can have other values.

I 've updated the unit test.
Please let me know if you have better ideas.

Comment (corona10) [reply Lib/test/test_posix.py]: Linux: http://man7.org/linux/man-pages/man2/sigaction.2.html
BSD: https://www.freebsd.org/cgi/man.cgi?query=siginfo&sektion=3&apropos=0&manpath=FreeBSD+7.1-RELEASE
Apple: https://opensource.apple.com/source/xnu/xnu-344/bsd/sys/signal.h

Is it enough to check `hasattr(signal, 'SIGCHLD')`?

File: Lib/test/test_posix.py [hunk 6]
```diff
@@ -1226,6 +1226,16 @@ def test_rename_dir_fd(self):
         finally:
             posix.close(f)
 
+    @unittest.skipUnless(hasattr(signal, 'SIGCHLD'), 'CLD_XXXX be placed in si_code for a SIGCHLD signal')
+    @unittest.skipUnless(hasattr(os, 'waitid_result'), "test needs os.waitid_result")
+    def test_cld_xxxx_constants(self):
+        self.assertTrue(hasattr(os, 'CLD_EXITED'))
```
Comment (serhiy-storchaka) [inline Lib/test/test_posix.py]: It is enough to just get an attribute:
```python
        os.CLD_EXITED
```
These constants are a part of Posix, but I am not sure that they are implemented in all Unixes (as you see, some was not implemented in old Linux). In particularly, I have doubts about AIX. But let try.

**Comment by corona10** [comment]
@serhiy-storchaka 
The documentation said that

> These are the possible values for si_code in the result returned by waitid().

I think that checking hasattr(os, 'waitid_result') and hasattr(signal, 'SIGCHLD') for unit test
is proper.

What do you think?

**Review by serhiy-storchaka — APPROVED** [review]

**Pull request merged**